registry/storage/blobserver.go

```diff
@@ -34,45 +34,45 @@ func (bs *blobServer) ServeBlob(ctx context.Context, w http.ResponseWriter, r *h
 		return err
 	}
 
-	redirectURL, err := bs.driver.URLFor(ctx, path, map[string]interface{}{"method": r.Method})
-
-	switch err.(type) {
-	case nil:
-		if bs.redirect {
+	if bs.redirect {
+		redirectURL, err := bs.driver.URLFor(ctx, path, map[string]interface{}{"method": r.Method})
+		switch err.(type) {
+		case nil:
 			// Redirect to storage URL.
 			http.Redirect(w, r, redirectURL, http.StatusTemporaryRedirect)
 			return err
-		}
 
-	case driver.ErrUnsupportedMethod:
-		// Fallback to serving the content directly.
-		br, err := newFileReader(ctx, bs.driver, path, desc.Size)
-		if err != nil {
+		case driver.ErrUnsupportedMethod:
+			// Fallback to serving the content directly.
```

Comment:
In the case of `driver.ErrUnsupportedMethod` we fall out of the type switch and then serve the content directly, so I think the comment is still valuable here.